linera-chain/src/unit_tests/chain_tests.rs

```diff
@@ -284,3 +286,191 @@ async fn test_application_permissions() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+/// Tests if services can execute as oracles if the total execution time is less than the limit.
+#[test_case(&[100]; "single service as oracle call")]
+#[test_case(&[50, 50]; "two service as oracle calls")]
+#[test_case(&[90, 10]; "long and short service as oracle calls")]
+#[test_case(&[33, 33, 33]; "three service as oracle calls")]
+#[tokio::test]
+async fn test_service_as_oracles(service_oracle_execution_times_ms: &[u64]) -> anyhow::Result<()> {
```

Comment:
Couldn't we merge the two test cases into one by accepting the `maximum_service_oracle_execution_ms` as the second argument to the test?

Reply:
We could, but I try to avoid having conditional assertions in tests, especially in unit tests. I try to stick to the "setup, execute, assert" pattern, and make these as obvious and streamlined as possible. I'd rather try to create some test helper functions to make the tests shorter instead. What do you think?

Reply:
If you prefer it that way - sure.

Reply:
Done. At least it's a start, we can improve this more later as well.